doc/md/actors-async.md

```diff
@@ -1,306 +0,0 @@
-
-To access the result of an `async` value, the receiver of the future use an `await` expression.
-
-For example, to use the result of `Counter.read()` above, we can first bind the future to an identifier `a`, and then `await a` to retrieve the underlying `Nat`, `n`:
-
-``` motoko include=counter
-let a : async Nat = Counter.read();
-let n : Nat = await a;
-```
-
-The first line immediately receives *a future of the counter value*, but does not wait for it, and thus cannot (yet) use it as a natural number.
-
-The second line `await`s this future and extracts the result, a natural number. This line may suspend execution until the future has been completed.
-
-Typically, one rolls the two steps into one and one just awaits an asynchronous call directly:
-
-``` motoko include=counter
-let n : Nat = await Counter.read();
-```
-
-Unlike a local function call, which blocks the caller until the callee has returned a result, a shared function call immediately returns a future, `f`, without blocking. Instead of blocking, a later call to `await f` suspends the current computation until `f` is complete. Once the future is completed (by the producer), execution of `await p` resumes with its result. If the result is a value, `await f` returns that value. Otherwise the result is some error, and `await f` propagates the error to the consumer of `await f`.
-
-Awaiting a future a second time will just produce the same result, including re-throwing any error stored in the future. Suspension occurs even if the future is already complete; this ensures state changes and message sends prior to *every* `await` are committed.
-
-:::danger
-
-A function that does not `await` in its body is guaranteed to execute atomically - in particular, the environment cannot change the state of the actor while the function is executing. If a function performs an `await`, however, atomicity is no longer guaranteed. Between suspension and resumption around the `await`, the state of the enclosing actor may change due to concurrent processing of other incoming actor messages. It is the programmer’s responsibility to guard against non-synchronized state changes. A programmer may, however, rely on any state change prior to the await being committed.
-
-:::
-
-For example, the implementation of `bump()` above is guaranteed to increment and read the value of `count`, in one atomic step. The alternative implementation:
-
-``` motoko no-repl
-  public shared func bump() : async Nat {
-    await inc();
-    await read();
-  };
-```
-
-does *not* have the same semantics and allows another client of the actor to interfere with its operation: each `await` suspends execution, allowing an interloper to change the state of the actor. By design, the explicit `await`s make the potential points of interference clear to the reader.
-
-## Traps and Commit Points
```

Comment:
It's important not to lose this section and traps and commit points. But I guess you are in the middle of moving stuff around

Reply:
Ah, I see it's in traps.md now. Nice!